src/components/Sidebar/Sidebar.js

```diff
@@ -0,0 +1,67 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+
+import Header from './components/Header';
+import NavList from './components/NavList';
+import NavItem from './components/NavItem';
+import Backdrop from './components/Backdrop';
+import CloseButton from './components/CloseButton';
+
+const SIDEBAR_WIDTH = 256;
+
+const baseStyles = ({ theme, open }) => css`
+  label: sidebar;
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  min-width: ${SIDEBAR_WIDTH}px;
+  background-color: ${theme.colors.n900};
+  transition: transform ${theme.transitions.default};
+  position: absolute;
+  transform: translateX(${open ? 0 : `-${SIDEBAR_WIDTH}px`});
```

Comment:
I see what you're doing here. 😊 I have two suggestions.

1. Can we pretty please move this into a separate style function? Ideally, we would be able to extract the style as a modifier BEM class, once we have static style extraction working for components (i.e., a class `sidebar--open`). We may also have to set `transform: translateX(0);` for the other breakpoints in that style function. I mean something like this:
```javascript
const openStyles = ({ open }) => open && css`
  label: sidebar--open;
  transform: translateX(-100%); // using 100% as per my next comment
  
  ${theme.mq.mega`
    transform: translateX(0);
  `;
`
```
2. I think this will have some smaller issues whenever the actual sidebar width is larger than `256`px. Further up, `min-width` of the sidebar is set to that value, so I understand it could be larger and the absolute value would not work here. Luckily, I think a `translateX(-100%)` would work just fine, right?

Please let me know if any of that was unclear. I'd happily explain some more.

Reply:
Great! That makes total sense!! I will add the `translate(-100%)` just now, thanks for point it out. One question, for what you said about the `openStyles`, there should be `transformX(0)` instead? Another thing, do you think that we should keep:

```
${theme.mq.mega`
    transform: translateX(0);
    position: relative;
  `};
```

on the `baseStyles` so we can always have it open on desktop devices besides the value of the `open` prop?